src/wp-includes/media.php

```diff
@@ -1059,24 +1059,40 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 		 */
 		$context = apply_filters( 'wp_get_attachment_image_context', 'wp_get_attachment_image' );
 
-		// Add `loading` attribute.
-		if ( wp_lazy_loading_enabled( 'img', $context ) ) {
-			$default_attr['loading'] = wp_get_loading_attr_default( $context );
-		}
-
 		$attr = wp_parse_args( $attr, $default_attr );
 
+		$loading_optimization_attr = wp_get_loading_optimization_attributes(
+			'img',
+			// $attr doesn't contain width and height by default.
+			array_merge(
+				array(
+					'width'  => $width,
+					'height' => $height,
+				),
+				$attr
+			),
+			$context
+		);
+
+		// Add loading optimization attributes if not available.
+		$attr = array_merge( $attr, $loading_optimization_attr );
```

Comment:
Normally we would use `wp_parse_args` here.

Reply:
`wp_parse_args()` doesn't have any benefit over `array_merge()` here, and it's probably slower since it's not a native PHP function.